### DISCUSSION TITLE
App still asks for email address when AUTH_REQUIRED is set to false

### DISCUSSION THREAD

**Discussion by PLP-GTR** [lukevella/rallly#649] [Q&A] [answered]
Hi!

I thought the parameter would turn this off.
I would like to have an easy-to-use instance for me and my friends without having anyone enter any email adresses.

Is this possible?
If not, what does the AUTH_REQUIRED parameter do?

Best regards,
Philipp

**Comment by lukevella** [comment]
Could be a bug. I’ll have a look when I get a chance. Can you try not setting the variable as it should default to false.

**Comment by PLP-GTR** [comment, marked as answer]
Answer: Expected behaviour (also check [docs](https://github.com/lukevella/rallly-selfhosted#4-secure-your-instance-optional))

lukevella:
> The reasons are mostly legacy at this point but it's still useful because as a guest it's very easy to lose access to your admin page unless you bookmark it which most people don't. By entering your email you will receive an link to the page in your inbox.

----

Yes, tested: Nothing changed. The app still asks for my email address:

![image](https://user-images.githubusercontent.com/3023230/229641473-5c56ef6f-46e7-447e-a766-68ebdfb5bf44.png)

My `config.env`:
```env
SECRET_PASSWORD=
NEXT_PUBLIC_BASE_URL=https://
SUPPORT_EMAIL=
SMTP_HOST=in-v3.mailjet.com
SMTP_PORT=465
SMTP_SECURE=true
SMTP_USER=
SMTP_PWD=
DISABLE_LANDING_PAGE=true
#AUTH_REQUIRED=false
```

After I enter any address I can use the app without verification (which is good in this case).

If I want to turn on notifications, I need to enter my address again, though. Imho it should be already stored:

![image](https://user-images.githubusercontent.com/3023230/229641645-893367d2-e19a-4a58-ba55-dbec62825e4c.png)

**Reply by lukevella** [reply]
Did you set a `SECRET_PASSWORD`? If not you need to set it to a random string that's at least 32 characters long.

**Reply by PLP-GTR** [reply]
Yes, of course :)

```
SECRET_PASSWORD=X...w=
NEXT_PUBLIC_BASE_URL=https://v...r.de
SUPPORT_EMAIL=support@v...r.de
SMTP_HOST=in-v3.mailjet.com
SMTP_PORT=465
SMTP_SECURE=true
SMTP_USER=c...d
SMTP_PWD=3...4
DISABLE_LANDING_PAGE=true
#AUTH_REQUIRED=false
```

**Reply by lukevella** [reply]
Ok, if you can share screen records of any odd behavior you're getting that would be helpful. I can't spot anything wrong from the information you've given me but there doesn't seem to be any issues relating to the `AUTH_REQUIRED` flag.

**Reply by PLP-GTR** [reply]
What is expected behaviour?
On step 3/3 of creating a new poll, the email is a required field.

**Reply by lukevella** [reply]
Try using rallly.co to see how it should work.

**Reply by PLP-GTR** [reply]
Ah, so expected behaviour is **required** email but **no required** verification.
So anyone can enter any email - why not make it obsolete? :)

**Reply by lukevella** [reply]
The reasons are mostly legacy at this point but it's still useful because as a guest it's very easy to lose access to your admin page unless you bookmark it which most people don't. By entering your email you will receive an link to the page in your inbox.

**Reply by PLP-GTR** [reply]
Ah, yes ... I understand.
That might be something for another discussion and your preferences.

My thoughts, just to think about, no need to change any behaviour: 
- Show a warning on step 3/3 if no mail is entered and show a warning on the poll (admin) page if no mail was entered
- Even if you lose the admin link, you can still participate because you shared the public link with others already
- Creating a new poll instead of changing it *because* having lost the admin link is fairly easy

**Reply by lukevella** [reply]
I agree, it will be possible to create a poll in the future without an email. There's a couple things in the way first but it will happen.
